src/backend_api.cpp

```diff
@@ -0,0 +1,268 @@
+        }
+        else
+        {
+            return miopenStatusBadParm;
+        }
+
+    case MIOPEN_ATTR_TENSOR_DIMENSIONS:
+        if(attributeType == MIOPEN_TYPE_INT64 && elementCount == 4)
+        {
+            m_Dimensions =
+                std::vector<int64_t>(reinterpret_cast<int64_t*>(arrayOfElements),
+                                     reinterpret_cast<int64_t*>(arrayOfElements) + elementCount);
+            m_DimensionsSet = true;
+            return miopenStatusSuccess;
+        }
+        else
+        {
+            return miopenStatusBadParm;
+        }
+
+    case MIOPEN_ATTR_TENSOR_STRIDES:
+        if(attributeType == MIOPEN_TYPE_INT64 && elementCount == 4)
+        {
+            m_Strides =
+                std::vector<int64_t>(reinterpret_cast<int64_t*>(arrayOfElements),
+                                     reinterpret_cast<int64_t*>(arrayOfElements) + elementCount);
+            m_StridesSet = true;
+            return miopenStatusSuccess;
+        }
+        else
+        {
+            return miopenStatusBadParm;
+        }
+
+    case MIOPEN_ATTR_TENSOR_VECTOR_COUNT:
+        if(attributeType == MIOPEN_TYPE_INT64 && elementCount == 1)
+        {
+            m_VectorCount    = *reinterpret_cast<int64_t*>(arrayOfElements);
+            m_VectorCountSet = true;
+            return miopenStatusSuccess;
+        }
+        else
+        {
+            return miopenStatusBadParm;
+        }
+
+    case MIOPEN_ATTR_TENSOR_VECTORIZED_DIMENSION:
+        if(attributeType == MIOPEN_TYPE_INT64 && elementCount == 1)
+        {
+            m_VectorizedDimension    = *reinterpret_cast<int64_t*>(arrayOfElements);
+            m_VectorizedDimensionSet = true;
+            return miopenStatusSuccess;
+        }
+        else
+        {
+            return miopenStatusBadParm;
+        }
+
+    case MIOPEN_ATTR_TENSOR_IS_VIRTUAL:
+        if(attributeType == MIOPEN_TYPE_BOOLEAN && elementCount == 1)
+        {
+            m_IsVirtual = *reinterpret_cast<bool*>(arrayOfElements);
+            return miopenStatusSuccess;
+        }
+        else
+        {
+            return miopenStatusBadParm;
+        }
+
+    case MIOPEN_ATTR_TENSOR_RAGGED_OFFSET_DESC:
+        if(attributeType == MIOPEN_TYPE_BACKEND_DESCRIPTOR && elementCount == 1)
+        {
+            m_RaggedOffsetDesc    = *reinterpret_cast<miopenBackendDescriptor_t*>(arrayOfElements);
+            m_RaggedOffsetDescSet = true;
+            return miopenStatusSuccess;
+        }
+        else
+        {
+            return miopenStatusBadParm;
+        }
+
+    default: return miopenStatusBadParm;
+    }
+}
+
+miopenStatus_t BackendTensorDescriptor::Finalize()
+{
+    if(!m_UniqueIdSet || !m_DataTypeSet || !m_ByteAlignmentSet || !m_DimensionsSet ||
+       !m_StridesSet || (m_VectorCountSet && !m_VectorizedDimensionSet))
+    {
+        return miopenStatusBadParm;
+    }
+
+    if(m_Dimensions.size() != 4 || m_Strides.size() != 4)
```

Comment:
Right now it is hardcoded to be 4d, should it be handled in a more common case?

Reply:
will remove it and rely on TensorDescriptor's layout checking